### ISSUE TITLE
Trouble Using Delivered-To

### ISSUE THREAD

**Issue by dstark** [dennisverspuij/tb-correctidentity#26]
Could you provide an example of how to use "Delivered-To" to trigger a particular identity? The situation I'm trying to address is as follows:

1. I'm a member of mailinglist@university.edu.
2. When I reply to a message from mailinglist@university.edu, I get an error from Thunderbird (not Correct Identity per se) to the effect that "A unique identity matching the From address was not found. The message will be sent using the current From field and settings from identity me@university.edu." So,
3. the reply starts off with From being mailinglist@university.edu. But
4. me@university.edu is in the Delivered-To header of the initial message, and that's the address I want to reply from to mailinglist@university.edu.

It looks like Correct Identity is able to handle this situation. But after some trial and a good amount of error, I'm coming up empty on precisely how.

I'm using Correct Identity 2.2.1 and Thunderbird 102.6.1.

Any guidance you could offer would be most appreciated. Thank you so much!

**Comment by ldreier** [comment]
Hi @dstark ,

sorry for the delay. Currently I am trying to set up some automatic tests for the addon and so I took your case as an example. But the test framework needed (and still needs) some work.
I also found a bug in the addon: When closing the configuration tab of the addon, you need to restart Thunderbird to make it work again. (Will be addressed in the next release).
So to your question:
I did some testing and it worked for me. Here are some screenshots to show the configuration:
I used Thunderbird 102.7.0 in the en-US Version.
A mail account was set up (test1@test.test) and "me@university.edu" was added as an additional identity to that account.
![1 account and identity](https://user-images.githubusercontent.com/30140713/213910003-2de01c2e-958b-4ecb-ac73-de3124271ac6.png)
For that account the the "find in reply" tickmark was checked and the "delivered-to" was added to the additional headers field (uppercase or lowercase is ignored here).
![2 selection settings](https://user-images.githubusercontent.com/30140713/213910004-5c1f0daf-4dac-41cc-b17f-cd068c105ab4.png)
![3 advanced settings](https://user-images.githubusercontent.com/30140713/213910006-7b6ab9a3-b3ba-4c7d-bcec-578ba832d103.png)
Then I put a test mail in the Inbox.
![4 testmail](https://user-images.githubusercontent.com/30140713/213910007-0f61a4b5-7554-4465-bf2b-637e1b9b654f.png)
When selecting "reply", correct identity replaced the "from" identity with "me@university.edu"
![5 reply](https://user-images.githubusercontent.com/30140713/213910008-33d823de-5882-4f1e-bd08-1ec31976a992.png)

So please check if this works for you. Or where do you think are significant differences to your setup.

**Comment by dstark** [comment]
@ldreier, thanks so much. That's most helpful. I follow most of what you've indicated pretty well and think I've replicated that on my system (Thunderbird 102.6.1; Correct Identity 2.2.1). But I'm still having replies default to the first To address rather than Delivered-To (although "delivered-to" is in the additional headers box).

What I note, though, that might be relevant is that, contrary to your last screenshot, the situation I have is

**- From: otheruser@university.edu
- To: mailinglist@university.edu**
- Delivered-To: me@university.edu

Would that possibly explain the behavior and require something a bit different in my setup?

**Comment by alexolog** [comment]
Here's an example based on the notification I got for this issue:
```
From: David Stark <notifications@github.com>
X-Resolved-to: my_name@email_provider.example
X-Delivered-to: github@my_domain.example
To: dennisverspuij/tb-correctidentity <tb-correctidentity@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
```
In this case, `X-Delivered-to` should take precedence, but I can envision situations when `To` or `CC`would be a better fit.
I would not want to use `X-Resolved-to` in any case.

**Comment by dstark** [comment]
Thanks so much @alexolog. I have been able to get the extension to change to the correct identity for an entirely different account by dropping x-original-to into the additional headers box because for that account in that situation, x-original-to is the header field that contains the proper identity.

x-original-to isn't in the mailinglist@university.edu headers at all. But somehow, when I add delivered-to to the additional headers box (thus: "x-original-to, delivered-to" or just "delivered-to" by itself), the from address still always defaults to the first mailing list in the prior message.

I'm not sure if this suggests any other ideas to you of possible resolutions. But the extension is incredibly helpful, so thank you so much for all that you've put into it!

**Comment by ldreier** [comment]
Fix is part of Release v2.3.2
